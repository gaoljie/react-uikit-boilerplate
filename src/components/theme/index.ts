import React from "react";

export const theme = {
  primary: "#EF6F35",
  primaryFont: "Montserrat",
  secondaryFont: "Work Sans",
  desktop: "@media (max-width: 1920px)",
  smallDesktop: "@media (max-width: 1441px)",
  tablet: "@media (max-width: 769px)",
  mobile: "@media (max-width: 600px)"
};

export const ThemeContext = React.createContext(theme);

export const ThemeProvider = ThemeContext.Provider;
