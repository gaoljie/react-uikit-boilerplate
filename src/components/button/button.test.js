import React from "react";
import "jest-styled-components";
import { shallow, mount } from "enzyme";
import Button from "./index";
import theme from "../theme";

describe("Test Button", () => {
  it("Matches the snapshot", () => {
    const wrapper = mount(<Button>button</Button>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Styling", () => {
    const wrapper = mount(<Button>button</Button>);

    expect(wrapper.find("button")).toHaveStyleRule(
      "background-color",
      theme.primary
    );
  });

  it("Render Prop", () => {
    const btnText = "Hello world";
    const wrapper = shallow(<Button>{btnText}</Button>);

    expect(wrapper.text()).toEqual(btnText);
  });

  it("Event Handler", () => {
    const mockFunction = jest.fn();
    const wrapper = shallow(<Button onClick={mockFunction}>Button</Button>);

    expect(mockFunction).not.toHaveBeenCalled();

    wrapper.simulate("click");

    expect(mockFunction).toHaveBeenCalled();
  });

  it("Disabled", () => {
    const mockFunction = jest.fn();
    const wrapper = mount(
      <Button onClick={mockFunction} disabled>
        Button
      </Button>
    );

    expect(mockFunction).not.toHaveBeenCalled();

    wrapper.simulate("click");

    expect(mockFunction).not.toHaveBeenCalled();
  });
});
