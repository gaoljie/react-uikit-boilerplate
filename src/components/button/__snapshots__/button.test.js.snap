// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button Matches the snapshot 1`] = `
.c0 {
  background-color: #EF6F35;
  font-family: Montserrat;
  font-weight: 600;
  padding: 15px 30px;
  color: white;
  font-size: 14px;
  border: 0;
  border-radius: 3px;
  cursor: pointer;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
}

.c0:hover {
  opacity: 0.8;
}

.c0[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
}

@media (max-width:600px) {
  .c0 {
    padding: 10px 15px;
  }
}

<Button>
  <Styled(styled.div)
    as="button"
  >
    <StyledComponent
      as="button"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              "
",
              "
  background-color: ",
              "#EF6F35",
              ";
  font-family: ",
              "Montserrat",
              ";
  font-weight: 600;
  padding: 15px 30px;
  color: white;
  font-size: 14px;
  border: 0;
  border-radius: 3px;
  cursor: pointer;
  outline: none;
  text-decoration: none;
  display: inline-block;
  &:hover {
    opacity: 0.8;
  }
  &[disabled] {
    cursor: not-allowed;
    opacity: 0.7;
  }

  ",
              "@media (max-width: 600px)",
              " {
    padding: 10px 15px;
  }
",
            ],
          },
          "displayName": "Styled(styled.div)",
          "foldedComponentIds": Array [
            "sc-bdVaJa",
          ],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <button
        className="c0"
      >
        button
      </button>
    </StyledComponent>
  </Styled(styled.div)>
</Button>
`;
